src/request.rs

```diff
@@ -78,12 +104,25 @@ pub trait KvRequest: Sync + Send + 'static + Sized {
 
     fn on_region_error(
         self,
-        _region_error: Error,
+        region_error: Error,
         pd_client: Arc<impl PdClient>,
+        backoff: Arc<impl Backoff>,
     ) -> BoxStream<'static, Result<Self::RpcResponse>> {
-        // Retry on region errors by default
-        // TODO: Add backoff and retry limit
-        self.response_stream(pd_client)
+        let mut backoff = *backoff;
```

Comment:
Here you do a memcpy of the backoff...
We can use `Box` instead of `Arc`, then `backoff` can be mutable.